File: J3.L.P0017/src/java/model/Gallery.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package model;

import java.util.List;

/**
 *
 * @author DuongDT
 */
public class Gallery {
    private int galleryCode;
    private String nameGallery;
    private String description;
    private String descriptionImage;
    private List<String> imageList;

    public Gallery() {
    }

    public Gallery(int galleryCode, String nameGallery) {
        this.galleryCode = galleryCode;
        this.nameGallery = nameGallery;
    }
    
    
    
    // contructor to get infomation of gallery, not neccessary to get all image in gallery
    public Gallery(int galleryCode, String nameGallery, String description, String descriptionImage) {
        this.galleryCode = galleryCode;
        this.nameGallery = nameGallery;
        this.description = description;
        this.descriptionImage = descriptionImage;
    }
    
    public Gallery(int galleryCode, String nameGallery, String description, String descriptionImage, List<String> imageList) {
        this.galleryCode = galleryCode;
        this.nameGallery = nameGallery;
        this.description = description;
        this.descriptionImage = descriptionImage;
        this.imageList = imageList;
    }

    public int getGalleryCode() {
        return galleryCode;
    }

    public String getNameGallery() {
        return nameGallery;
    }

    public String getDescription() {
            return description;
    }

    public String getDescriptionImage() {
        return descriptionImage;
    }

    public List<String> getImageList() {
        return imageList;
    }
    
    
}

File: J3.L.P0017/web/js/gallery.js
/* 
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
    
    // Function: Show detail image when user click on image card.
   
    // Get list of all images in slide.
    const image= document.querySelectorAll(".image-slide");
    // Get list of all images card
    const imageCard = document.querySelectorAll(".image-card");
    
    // In the firs time access gallery page, the first image on slide will be display
    display(0);
    
    //With any image card element, add event on click. 
    imageCard.forEach(addEventForCard);
    
    function addEventForCard(item, index) {
        item.addEventListener("click",function () {
            display(index);
        });
    }
    // Hiden all images in slide.
    function hideImage() {
        for (let j=0;j<image.length;j++) {
            // If status of image is display, hiden it.
            if (image[j].classList.contains("display-image")) {
                image[j].classList.remove("display-image");
                image[j].classList.add("hiden-image");
            }
        }
    }
    //Display an image into slide  
    function display (imageIndex)  {
        hideImage();
        image[imageIndex].classList.remove("hiden-image");
        image[imageIndex].classList.add("display-image");
    }

    

    
    